session/refresh.go

```diff
@@ -0,0 +1,80 @@
+package session
+
+import (
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/g8rswimmer/go-sfdc"
+)
+
+// Refresher is a structure will allow for a session token to be
+// re-authenticated based on a time interval.
+type Refresher struct {
+	session     *Session
+	sessionLock sync.RWMutex
+	err         error
+	stop        chan struct{}
+}
+
+// OpenRefresh will open the first session and set up a re-occuring refresh of the session token.
+func OpenRefresh(config sfdc.Configuration, refershTime int) (*Refresher, error) {
+	if refershTime <= 0 {
+		return nil, errors.New("session refresh: refresh time can not be less than zero")
+	}
+	session, err := Open(config)
+	if err != nil {
+		return nil, err
+	}
+	refresher := &Refresher{
+		session: session,
+		stop:    make(chan struct{}),
+	}
+
+	go func() {
+		sleep := time.Duration(refershTime) * time.Second
+		for {
+			select {
+			case <-refresher.stop:
+				return
+			default:
+			}
+			time.Sleep(sleep)
+			if err := refresher.refresh(); err != nil {
+				refresher.err = err
+				sleep = 5 * time.Second
```

Comment:
not sure I follow the 5 second hard-code here, can you explain?

Reply:
that was just a magic number, any suggestions?

Reply:
tried to handle with a user supplied struct (also added a default one because I am nice)